device/api/umd/device/chip/chip.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2024 Tenstorrent Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "umd/device/tt_cluster_descriptor_types.h"
+#include "umd/device/tt_soc_descriptor.h"
+
+namespace tt::umd {
+
+// An abstract class that represents a chip.
+class Chip {
+public:
+    Chip(tt_SocDescriptor soc_descriptor);
+
+    virtual ~Chip() = default;
+
+    tt_SocDescriptor& get_soc_descriptor();
+
+    virtual bool is_mmio_capable() const = 0;
+
+private:
+    tt_SocDescriptor soc_descriptor_;
```

Comment:
I have no idea. It will be an easy fix nevertheless.
I plan to propose something and then to figure out a way to come to an agreement, and then implement the agreed changes in the whole codebase.

Reply:
makes sense